src/rules/enforce-boolean-naming-prefixes.ts

```
@@ -160,11 +160,64 @@ export const enforceBooleanNamingPrefixes = createRule<Options, MessageIds>({
+        // 2. It's not used with array/object literals as fallbacks
+        if (
+          node.init.type === AST_NODE_TYPES.LogicalExpression &&
+          node.init.operator === '||'
+        ) {
+          // Check if right side is a non-boolean literal (array, object, string, number)
+          const rightSide = node.init.right;
+          if (
+            rightSide.type === AST_NODE_TYPES.ArrayExpression ||
+            rightSide.type === AST_NODE_TYPES.ObjectExpression ||
+            (rightSide.type === AST_NODE_TYPES.Literal &&
+             typeof rightSide.value !== 'boolean')
```

ESLint

[error] Insert `·`

(prettier/prettier)